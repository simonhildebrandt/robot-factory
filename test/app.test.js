import 'babel-polyfill'

import {App} from '../client/index.tsx'
import React from 'react'
import axios from 'axios'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'


configure({adapter: new Adapter()})

jest.mock('axios')
axios.get.mockResolvedValue({data: [
  {
    "id": 1,
    "name": "robot 1",
    "configuration": {
      "hasSentience": true,
      "hasWheels": true,
      "hasTracks": true,
      "numberOfRotors": 1,
      "Colour": "red"
    },
    "statuses": ["on fire", "rusty", "loose screws", "paint scratched"]
  }
]});


test('App render starting state', () => {
  const component = shallow(<App/>)
  expect(component).toMatchSnapshot()
});

test('App render loaded state', async () => {
  const component = await shallow(<App/>)
  component.update()
  expect(component).toMatchSnapshot()
});
