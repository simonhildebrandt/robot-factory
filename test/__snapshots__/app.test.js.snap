// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render 1`] = `
<div>
  <h1>
    Robot Factory
  </h1>
  <em>
    loading robots...
  </em>
</div>
`;
